src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPostBuildStatus.java

```diff
@@ -28,27 +29,37 @@ public BitbucketPostBuildStatus(String serverId) {
         this(serverId, new DefaultJenkinsProvider());
     }
 
-    @Override
-    public void decorateCheckoutCommand(GitSCM scm, Run<?, ?> run, GitClient git, TaskListener listener,
-                                        CheckoutCommand cmd) throws GitException {
+    public Revision decorateRevisionToBuild(GitSCM scm, Run<?, ?> run, GitClient git, TaskListener listener, Revision marked, Revision rev) throws GitException {
         Injector injector = jenkinsProvider.get().getInjector();
         if (injector == null) {
             listener.getLogger().println("Injector could not be found while creating build status");
-            return;
+            return rev;
         }
 
         BuildData buildData = scm.getBuildData(run);
+
+        String ref = rev.getBranches().stream()
+                .map(branch -> {
+                    String name = branch.getName();
+                    if (StringUtils.startsWith(name, "refs/")) {
+                        return name;
+                    }
+                    return "refs/heads/" + name;
```

Comment:
Should we do some defensive checks, e.g. `isBlank(name)`?

Reply:
No need, the stringUtils startsWith above handles that (which is why I picked the stringutils one, rather than the built in java one)

Reply:
oh yeah, duh 🤦‍♀